CorsixTH/Lua/dialogs/fullscreen/staff_management.lua

```diff
@@ -537,18 +539,27 @@ function UIStaffManagement:payBonus()
     staff:changeAttribute("happiness", 0.5)
     self.hospital:spendMoney(math_floor(staff.profile.wage*0.1), _S.transactions.personal_bonus)
     self.ui:playSound("cashreg.wav")
+  else
+    self.ui:playSound("wrong2.wav")
   end
 end
 
 function UIStaffManagement:increaseSalary()
   if self.selected_staff then
```

Comment:
Function is growing - let's play wrong and exit early on no staff rather than indenting everything

Reply:
So:
```
if not self.selected_staff then
  self.ui:playSound("wrong2.wav")
  return
end
```

Reply:
yup

Reply:
Other option is to do the good weather case and return, then close all "if" statements, and just before the end of the function play the "wrong" sound. In that way you don't get multiple calls to the same "wong" sound at the end.

ie
```
func ...
  if self.selected staff
    ...
    if (staff: increaseWage(...) then
      play happy sound
      return
    end
  end

  play wrong sound
end
```

Reply:
I've used Alberth's version

Reply:
A bit of it is that testing for a condition and reacting to it are separate concerns that ideally wouldn't be in the same function. Multiple `return false` are acceptable in a way that multiple playSound/return lines are not.

But thats a level of standard I don't expect.